hyperspy/io_plugins/semper_unf.py

```diff
@@ -0,0 +1,563 @@
+                    ('ICLAYN', ('<i2', 2)),   # Bytes 17,18
+                    ('ICLASS', '<i2'),        # Bytes 19
+                    ('IFORM', '<i2'),         # Bytes 20
+                    ('IWP', '<i2'),           # Bytes 21
+                    ('DATE', ('<i2', 6)),     # Bytes 22-27
+                    ('NCRANG', '<i2'),        # Bytes 28
+                    ('RANGE', ('<i2', 27)),   # Bytes 29-55
+                    ('IPLTYP', '<i2'),        # Bytes 56
+                    ('NCOLH', '<i2'),         # Bytes 57
+                    ('NROWH', '<i2'),         # Bytes 58
+                    ('NLAYH', '<i2'),         # Bytes 59
+                    ('ICCOLNH', '<i2'),       # Bytes 60
+                    ('ICROWNH', '<i2'),       # Bytes 61
+                    ('ICLAYNH', '<i2'),       # Bytes 62
+                    ('REALCO', '<i2'),        # Bytes 63
+                    ('NBLOCK', '<i2'),        # Bytes 64
+                    ('FREE', ('<i2', 3)),     # Bytes 65-67
+                    ('DATAV6', ('<i2', 4)),   # Bytes 68-71
+                    ('DATAV7', ('<i2', 4)),   # Bytes 72-75
+                    ('DZV5', ('<i2', 4)),     # Bytes 76-79
+                    ('Z0V4', ('<i2', 4)),     # Bytes 80-83
+                    ('DYV3', ('<i2', 4)),     # Bytes 84-87
+                    ('Y0V2', ('<i2', 4)),     # Bytes 88-91
+                    ('DXV1', ('<i2', 4)),     # Bytes 92-95
+                    ('X0V0', ('<i2', 4)),     # Bytes 96-99
+                    ('NTITLE', '<i2'),        # Bytes 100
+                    ('TITLE', ('<i2', 144)),  # Bytes 101-244
+                    ('XUNIT', ('<i2', 4)),    # Bytes 245-248
+                    ('YUNIT', ('<i2', 4)),    # Bytes 249-252
+                    ('ZUNIT', ('<i2', 4))]    # Bytes 253-256
+
+    def __init__(self, data, title=Undefined, offsets=(0., 0., 0.), scales=(1., 1., 1.),
+                 units=(Undefined, Undefined, Undefined), metadata=None):
+        self._log.debug('Calling __init__')
+        if metadata is None:
+            metadata = {}
+        data = data[tuple(None for _ in xrange(3-len(data.shape)))]  # Make sure data is 3D!
+        self.data = data
+        self.title = title
+        self.offsets = offsets
+        self.scales = scales
+        self.units = units
+        self.metadata = metadata
+        self._log.debug('Created '+str(self))
+
+    @classmethod
+    def _read_label(cls, unf_file):
+        rec_length = np.fromfile(unf_file, dtype='<i4', count=1)[0]  # length of label
+        label = sarray2dict(np.fromfile(unf_file, dtype=cls.LABEL_DTYPES, count=1))
+        label.update({'SEMPER': ''.join([str(unichr(l)) for l in label['SEMPER']])})
+        assert label['SEMPER'] == 'Semper'
+        # Process dimensions:
+        for key in ['NCOL', 'NROW', 'NLAY', 'ICCOLN', 'ICROWN', 'ICLAYN']:
+            value = 256**2*label.pop(key+'H') + 256*label[key][0] + label[key][1]
+            label.update({key: value})

```

Comment:
Why not just `label[key] = value`? I suppose it's the same, but just seems a little strange. It would be slightly quicker as well:
```python
In [3]: %timeit {}.update({'a':1})
The slowest run took 6.20 times longer than the fastest. This could mean that an intermediate result is being cached 
1000000 loops, best of 3: 308 ns per loop

In [4]: %timeit {}['a'] = 1
10000000 loops, best of 3: 71.7 ns per loop
```